### ISSUE TITLE
POM-Editor should open pom's in local Maven-repo as read-only

### ISSUE THREAD

**Issue by HannesWell** [eclipse-m2e/m2e-core#751]
When opening the pom of a dependency in the local maven cache, for example by Ctrl+Clicking on a corresponding dependency element, the POM-Editor allows modifications of that POM. But since the pom's in the local-maven cache is only intended to be modified by corresponding downloads from another repo or installs, such POM should be opened read-only. Of course one could still open the POM in a text-editor and modify it there, but we should not support such hacky actions, especially because it can happen unintended.

This is likely an upstream-issue but I'm not sure which project it is exactly?
@mickaelistria you can probably tell best.

**Comment by mickaelistria** [comment]
The ctrl+click target is provided by the language server; and LSP doesn't make a distinction of read-only or not target.
So the smartness should be implemented somewhere in m2e that should load a non-modifiable document for those files. However, I don't know how to implement that reliably. There used to be some code in m2e that was opening links in read-only mode in some cases, but it was more tied to the editor than to the document, so it wouldn't fit for the LSP-based story.

**Comment by laeubi** [comment]
I think one can assume each file as read-only that can't be located in the workspace. Then show a "Enable edit" somewhere with a warning that the file is not part of the workspace?
Another option would be to fetch the current local repository path and disable editing for any file resing inside this folder.

**Comment by mickaelistria** [comment]
> I think one can assume each file as read-only that can't be located in the workspace.

No. Eclipse IDE has "File > Open File" to open a file in the IDE, most likely to edit it. So in general, any file opened in the IDE might be opened in order to be edited. The read-only case is more the corner case.

> Another option would be to fetch the current local repository path and disable editing for any file resing inside this folder.

That sounds safer, and at least it's a fix in m2e for an m2e-specific problem.
